python/benchmarks/finalize_staged_data.py

```diff
@@ -61,45 +65,51 @@ def setup(self, cache:CachedDFGenerator, param:int):
         self.lib.write(self.symbol, data=df, prune_previous_versions=True)
         stage_chunks(self.lib, self.symbol, cachedDF, INITIAL_TIMESTAMP, list_of_chunks)
 
-    def time_finalize_staged_data(self, cache:CachedDFGenerator, param:int):
+    def time_finalize_staged_data(self, cache: CachedDFGenerator, param: int):
         print(">>> Library:", self.lib)
         print(">>> Symbol:", self.symbol)
         self.lib.finalize_staged_data(self.symbol, mode=StagedDataFinalizeMethod.WRITE)
 
-    def peakmem_finalize_staged_data(self, cache:CachedDFGenerator, param:int):
+    def peakmem_finalize_staged_data(self, cache: CachedDFGenerator, param: int):
         print(">>> Library:", self.lib)
         print(">>> Symbol:", self.symbol)
         self.lib.finalize_staged_data(self.symbol, mode=StagedDataFinalizeMethod.WRITE)
 
-    def teardown(self, cache:CachedDFGenerator, param:int):
+    def teardown(self, cache: CachedDFGenerator, param: int):
         self.ac.delete_library(self.lib_name)
 
+
+from asv_runner.benchmarks.mark import SkipNotImplemented
+
+
 class FinalizeStagedDataWiderDataframeX3(FinalizeStagedData):
-    '''
-        The test is meant to be executed with 3 times wider dataframe than the base test
-    '''
+    """
+    The test is meant to be executed with 3 times wider dataframe than the base test
+    """
 
     def setup_cache(self):
         # Generating dataframe with all kind of supported data type
-        cachedDF = CachedDFGenerator(350000, [5, 25, 50]) # 3 times wider DF with bigger string columns
+        cachedDF = CachedDFGenerator(
+            350000, [5, 25, 50]
+        )  # 3 times wider DF with bigger string columns
         return cachedDF
-    
-    def setup(self, cache:CachedDFGenerator, param:int):
-        if (not SLOW_TESTS):
-            raise SkipNotImplemented ("Slow tests are skipped")
-        super().setup(cache,param)
-
-    def time_finalize_staged_data(self, cache:CachedDFGenerator, param:int):
-        if (not SLOW_TESTS):
-            raise SkipNotImplemented ("Slow tests are skipped")
-        super().time_finalize_staged_data(cache,param)
-
-    def peakmem_finalize_staged_data(self, cache:CachedDFGenerator, param:int):
-        if (not SLOW_TESTS):
-            raise SkipNotImplemented ("Slow tests are skipped")
-        super().peakmem_finalize_staged_data(cache,param)
-
-    def teardown(self, cache:CachedDFGenerator, param:int):
-        if (SLOW_TESTS):
+
+    def setup(self, cache: CachedDFGenerator, param: int):
+        if not SLOW_TESTS:
```

Comment:
Not your code but wouldn't it be cleaner to lift the `SLOW_TESTS` conditions up to control whether we define the class at all? Rather than having them on each method?

Reply:
I am keen on merging this as is because I want to rebase my numpy branch with these changes, so I've created a separate issue (#2070) to track this and fix it in the future.